lib/elm-test.js

```diff
@@ -444,3 +327,202 @@ ${testFileGlobs.join('\n')}
 Are the above patterns correct? Maybe try running elm-test with no arguments?
       `.trim();
 }
+
+const parsePositiveInteger = (flag /*: string */) => (
+  string /*: string */
+) /*: number */ => {
+  const number = Number(string);
+  if (!/^\d+$/.test(string)) {
+    console.error(
+      `error: option '${flag}' expected one or more digits, but got: ${string}`
+    );
+    throw process.exit(1);
+  } else if (!Number.isFinite(number)) {
+    console.error(
+      `error: option '${flag}' expected a finite number, but got: ${number}`
+    );
+    throw process.exit(1);
+  } else {
+    return number;
+  }
+};
+
+const parseReport = (flag /*: string */) => (
+  string /*: string */
+) /*: typeof Report.Report */ => {
+  const result = Report.parse(string);
+  switch (result.tag) {
+    case 'Ok':
+      return result.value;
+    case 'Error':
+      console.error(`error: option '${flag}' ${result.error}`);
+      throw process.exit(1);
```

Comment:
The next version of Commander has built-in support for validating sets of valid strings, which would help here. https://github.com/tj/commander.js/issues/518#issuecomment-716094575